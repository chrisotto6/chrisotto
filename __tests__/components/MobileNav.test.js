import { render, fireEvent } from '@testing-library/react'
import MobileNav from '@/components/MobileNav'

it('renders mobile nav unchanged', () => {
  const { container } = render(<MobileNav />)
  expect(container).toMatchSnapshot()
})

it('button handles click correctly', () => {
  const { getByTestId } = render(<MobileNav />)
  fireEvent.click(getByTestId('mobile-nav-button'))
  expect(document.body.style.overflow).toBe('hidden')
  fireEvent.click(getByTestId('mobile-nav-button'))
  expect(document.body.style.overflow).toBe('auto')
})
