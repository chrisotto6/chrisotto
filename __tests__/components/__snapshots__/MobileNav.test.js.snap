// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders mobile nav unchanged 1`] = `
<div>
  <div
    class="sm:hidden"
  >
    <button
      aria-label="Toggle Menu"
      class="ml-1 mr-1 h-8 w-8 rounded py-1"
      data-testid="mobile-nav-button"
      type="button"
    >
      <svg
        class="text-gray-900 dark:text-gray-100"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
    <div
      class="fixed top-24 right-0 z-10 h-full w-full transform bg-gray-200 opacity-95 duration-300 ease-in-out dark:bg-gray-800 translate-x-full"
    >
      <button
        aria-label="toggle modal"
        class="fixed h-full w-full cursor-auto focus:outline-none"
        type="button"
      />
      <nav
        class="fixed mt-8 h-full"
      >
        <div
          class="px-12 py-4"
        >
          <a
            class="text-2xl font-bold tracking-widest text-gray-900 dark:text-gray-100"
            href="/blog"
          >
            Blog
          </a>
        </div>
        <div
          class="px-12 py-4"
        >
          <a
            class="text-2xl font-bold tracking-widest text-gray-900 dark:text-gray-100"
            href="/tags"
          >
            Tags
          </a>
        </div>
        <div
          class="px-12 py-4"
        >
          <a
            class="text-2xl font-bold tracking-widest text-gray-900 dark:text-gray-100"
            href="/projects"
          >
            Projects
          </a>
        </div>
        <div
          class="px-12 py-4"
        >
          <a
            class="text-2xl font-bold tracking-widest text-gray-900 dark:text-gray-100"
            href="/about"
          >
            About
          </a>
        </div>
        <div
          class="px-12 py-4"
        >
          <a
            class="text-2xl font-bold tracking-widest text-gray-900 dark:text-gray-100"
            href="/uses"
          >
            Uses
          </a>
        </div>
      </nav>
    </div>
  </div>
</div>
`;
